pkg/controller/job/job_controller.go

```diff
@@ -893,9 +908,10 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
 		}
 	}
 
-	needsStatusUpdate := suspendCondChanged || active != job.Status.Active || !equalReady(ready, job.Status.Ready)
+	needsStatusUpdate := suspendCondChanged || terminatingPods != job.Status.Terminating || active != job.Status.Active || !equalReady(ready, job.Status.Ready)
```

Comment:
`pointer.Int32Equal`?


Could `equalReady` go away actually?

Reply:
Good call out. I opened up https://github.com/kubernetes/kubernetes/pull/119111 to address `equalReady`.

Reply:
Updated this also.